File: leetcode/src/main/java/leetcode/number/PowerofFour.java
package leetcode.number;

// https://leetcode.com/problems/power-of-four/
public class PowerofFour {

	public static boolean isPowerOfFour(int n) {
		if (n <= 0) {
			return false;
		}
		while (n > 1) {
			if (n % 4 == 0) {
				n /= 4;
			} else {
				return false;
			}
		}
		return true;
	}

	public static boolean isPowerOfAny(int n, int k) {
		if (n <= 0) {
			return false;
		}
		while (n > 1) {
			if (n % k == 0) {
				n /= k;
			} else {
				return false;
			}
		}
		return true;
	}

	public static void main(String[] args) {
		System.out.println(isPowerOfFour(8));
	}

}

File: py-algorithm/number/power_of_four.py
import unittest

"""
    https://leetcode.com/problems/power-of-four/
    
    Kiểm tra số n có phải là lũy thừa của 4 hay không?
"""
class Solution(unittest.TestCase):

    def isPowerOfFour(self, n: int) -> bool:
        if n == 0: return False
        while n % 4 == 0:
            n //= 4
        return n == 1

    def test(self):
        self.assertEqual(True, self.isPowerOfFour(1))
        self.assertEqual(True, self.isPowerOfFour(16))


if __name__ == '__main__':
    unittest.main()
